x/stakeibc/keeper/icqcallbacks_delegator_shares.go

```diff
@@ -62,28 +62,35 @@ func DelegatorSharesCallback(k Keeper, ctx sdk.Context, args []byte, query icqty
 		return errorsmod.Wrapf(types.ErrValidatorNotFound, "no registered validator for address (%s)", queriedDelegation.ValidatorAddress)
 	}
 
-	// Check if the ICQ overlapped a delegation or undelegation ICA that would have modfied the number of delegated tokens
+	// Check if the ICQ overlapped a delegation, undelegation, or detokenization ICA
+	// that would have modfied the number of delegated tokens
 	prevInternalDelegation := callbackData.InitialValidatorDelegation
 	currInternalDelegation := validator.Delegation
-	icaOverlappedIcq, err := k.CheckDelegationChangedDuringQuery(ctx, chainId, prevInternalDelegation, currInternalDelegation)
+	icaOverlappedIcq, err := k.CheckDelegationChangedDuringQuery(
+		ctx,
+		chainId,
+		validator.Address,
+		prevInternalDelegation,
+		currInternalDelegation,
+	)
 	if err != nil {
 		return err
 	}
 
+	// If the ICA/ICQ overlapped, submit a new query
 	if icaOverlappedIcq {
-		// If the ICA/ICQ overlapped, submit a new query
 		if err := k.InterchainQueryKeeper.RetryICQRequest(ctx, query); err != nil {
 			return errorsmod.Wrapf(err, "unable to resubmit delegator shares query")
 		}
 		return nil
-	} else {
-		// If there was no ICA/ICQ overlap, update the validator to indicate that the query
-		//  is no longer in progress (which will unblock LSM liquid stakes to that validator)
-		validator.SlashQueryInProgress = false
-		hostZone.Validators[valIndex] = &validator
-		k.SetHostZone(ctx, hostZone)
 	}
 
+	// If there was no ICA/ICQ overlap, update the validator to indicate that the query
+	//  is no longer in progress (which will unblock LSM liquid stakes to that validator)
+	validator.SlashQueryInProgress = false
+	hostZone.Validators[valIndex] = &validator
+	k.SetHostZone(ctx, hostZone)
```

Comment:
Not for this PR, but this is evidence we should do the refactor to change the way validators are stored in a prefix store. Just to update a property on the validator takes 3 lines of code is asking for mistakes to be made.

Reply:
amen to that